lib/galaxy/managers/sharable.py

```diff
@@ -163,9 +283,12 @@ def share_with(self, item, user, flush=True):
         Get or create a share for the given user (or users if `user` is a list).
         """
         # precondition: user has been validated
-        # allow user to be a list and call recursivly
+        # allow user to be a list and call recursively
         if isinstance(user, list):
-            return [self.share_with(item, _, flush=False) for _ in user]
+            user_assocs = [self.share_with(item, _, flush=False) for _ in user]
```

Comment:
If the user who shares writes multiple emails separated by `,` in the UI, this can be an actual list of users:
https://github.com/OlegZharkov/galaxy/commit/0fa8470bd86be7db6f79560a9a30bbf1958b4880#diff-f0de4cf20668343a55f08ea7b70f0e138ee9cbe891e637ba359594422d54fafeR719

Reply:
With the new approach of sharing and unsharing with a single list, this is no longer needed so the code has been removed in https://github.com/galaxyproject/galaxy/pull/11701/commits/508f96b65ae65c250ed77f07b13a09dc3620427c as suggested.
Thanks @mvdbeek!